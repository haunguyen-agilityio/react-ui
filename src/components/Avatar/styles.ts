// lib MUI
import { Avatar } from '@mui/material'
import { styled } from '@mui/system'

// theme
import theme from '../../themes'

export const StyledAvatar = styled(Avatar)({
  backgroundColor: '#21d083',
  fontWeight: 600,
  color: theme.palette.common.white,
})

export const SmallAvatar = styled(Avatar)({
  backgroundColor: '#aaaaaa',
})
