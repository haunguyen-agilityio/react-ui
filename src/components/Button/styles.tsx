// libraries
import { styled } from '@mui/system'

// components
import { Button } from '@mui/material'

// theme
import typography from '../../themes/typography'

const StyledButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(),
  padding: `${theme.spacing(1.7)} ${theme.spacing(3)}`,
  height: '38px',
  borderRadius: 0,
  boxShadow: 'none',
  textTransform: 'capitalize',
  fontSize: typography.h4.fontSize,
  color: theme.palette.common.white,
  '&:hover': {
    boxShadow: 'none',
  },
  '&.Mui-disabled': {
    color: theme.palette.common.black,
    backgroundColor: theme.palette.primary.main,
    opacity: 0.5,
  },
  '&.MuiButton-containedSecondary': {
    color: theme.palette.common.black,
    backgroundColor: '#eaeaea',
    border: '1px solid #cbcbcb',
    '&:hover': {
      backgroundColor: '#d0d0d0',
    },
    '&.Mui-disabled': {
      backgroundColor: '#eaeaea',
    },
  },
  '.MuiSvgIcon-root': {
    width: '12px',
    height: '12px',
  },
}))

export default StyledButton
