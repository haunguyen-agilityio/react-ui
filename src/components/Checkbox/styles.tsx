// libraries
import { Checkbox, CheckboxProps, FormControlLabel } from '@mui/material'
import { styled } from '@mui/system'

export const StyledFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
  '.MuiFormControlLabel-label.Mui-disabled': {
    color: theme.palette.common.black,
  },
}))

interface ICheckboxProps extends CheckboxProps {
  isError?: boolean
}

export const StyledCheckbox = styled((props: ICheckboxProps) => <Checkbox {...props} />, {
  shouldForwardProp: (prop) => prop !== 'isError',
})(({ theme, isError }) => ({
  color: isError ? theme.palette.error.main : '#dedede',
  '&:hover': {
    backgroundColor: 'transparent',
  },
  '&.Mui-disabled': {
    opacity: 0.2,
  },
  '.MuiSvgIcon-root': {
    transform: 'scale(1.25)',
  },
  '&.Mui-checked': {
    color: theme.palette.primary.main,
  },
}))
