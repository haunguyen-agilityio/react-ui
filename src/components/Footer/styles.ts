// Libraries
import { styled } from '@mui/system'
import ListItemButton from '@mui/material/ListItemButton'

// Themes
import theme from '../../themes'

export const StyledListItemButton = styled(ListItemButton)({
  textAlign: 'center',
  color: theme.palette.common.white,
  marginRight: theme.spacing(0.5),
  padding: theme.spacing(1.5, 0),
  '&:hover': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
  '&.Mui-selected': {
    backgroundColor: theme.palette.grey[400],
    color: theme.palette.common.black,
  },
})

export const StyledFooter = styled('footer')({
  color: theme.palette.common.white,
  backgroundColor: theme.palette.grey[500],
  padding: theme.spacing(0, 4.2),
})
