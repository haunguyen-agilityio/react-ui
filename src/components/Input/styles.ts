// Libraries
import { styled } from '@mui/system'
import TextField from '@mui/material/TextField'
import FormHelperText from '@mui/material/FormHelperText'

// Themes
import theme from '@themes'
import typography from '@themes/typography'

export const InputWrapper = styled('div')({
  padding: theme.spacing(2),
})

export const StyledFormHelperText = styled(FormHelperText)({
  color: theme.palette.error.main,
  fontSize: typography.body1.fontSize,
})

export const StyledTextField = styled(TextField)((props) => ({
  '& .MuiInputBase-input': {
    color: props.error ? theme.palette.error.main : theme.palette.text.primary,
  },
  ' .MuiOutlinedInput-root': {
    '& fieldset': {
      borderRadius: 0,
      padding: theme.spacing(0),
      borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      '&.MuiOutlinedInput-notchedOutline': {
        borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      },
    },
    '&:hover fieldset': {
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
    '&.Mui-focused fieldset': {
      borderColor: theme.palette.grey[400],
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
  },
}))
