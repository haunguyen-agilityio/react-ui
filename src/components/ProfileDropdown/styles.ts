// Libraries
import { styled } from '@mui/system'
import Accordion from '@mui/material/Accordion'
import AccordionSummary from '@mui/material/AccordionSummary'
import { MenuItem, MenuList, Link } from '@mui/material'

// Themes
import theme from '@themes'

export const DropdownStyles = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  width: '100%',
})

export const StyledAccordion = styled(Accordion)({
  width: theme.spacing(26.8),
  padding: theme.spacing(1.6),
  fontSize: theme.spacing(1.6),
})

export const StyledAccordionSummary = styled(AccordionSummary)({
  border: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
})

export const StyledMenuItem = styled(MenuItem)({
  borderTop: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
  padding: theme.spacing(1.6),
})

export const StyledMenuList = styled(MenuList)({
  width: theme.spacing(30),
  padding: theme.spacing(0),
})

export const StyledLink = styled(Link)({
  textDecoration: 'none',
  color: theme.palette.text.primary,
  fontSize: theme.spacing(1.6),
})
