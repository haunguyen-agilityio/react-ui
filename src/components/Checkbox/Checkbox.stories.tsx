// libraries
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { ThemeProvider } from '@mui/material/styles'

// components
import Checkbox from '@components/Checkbox'

// themes
import theme from '@themes'

export default {
  component: Checkbox,
  title: 'Components/Checkbox',
} as ComponentMeta<typeof Checkbox>

const Template: ComponentStory<typeof Checkbox> = (args) => (
  <ThemeProvider theme={theme}>
    <Checkbox {...args} />
  </ThemeProvider>
)

export const Default = Template.bind({})
Default.args = {
  label: 'Default',
  onChange: () => {},
}

export const Checked = Template.bind({})
Checked.args = {
  label: 'Default',
  isChecked: true,
  onChange: () => {},
}

export const Disabled = Template.bind({})
Disabled.args = {
  label: 'Disabled',
  onChange: () => {},
  disable: true,
}

export const Error = Template.bind({})
Error.args = {
  label: 'Error',
  onChange: () => {},
  isError: true,
}
