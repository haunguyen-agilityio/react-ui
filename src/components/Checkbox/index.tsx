// libraries
import SquareIcon from '@mui/icons-material/Square'
import SquareOutlinedIcon from '@mui/icons-material/SquareOutlined'
import { memo } from 'react'

// styles
import { StyledFormControlLabel, StyledCheckbox } from './styles'

interface ICheckboxProps {
  label: string
  isChecked: boolean
  onChange: () => void
  disable?: boolean
  isError?: boolean
}

const Checkbox = ({
  label = 'Label',
  disable = false,
  onChange = () => {},
  isError = false,
  isChecked = false,
}: ICheckboxProps) => {
  return (
    <StyledFormControlLabel
      control={
        <StyledCheckbox
          isError={isError}
          icon={<SquareOutlinedIcon />}
          checkedIcon={<SquareIcon />}
          checked={isChecked}
        />
      }
      label={label}
      disabled={disable}
      onChange={onChange}
    />
  )
}

export default memo(Checkbox)
