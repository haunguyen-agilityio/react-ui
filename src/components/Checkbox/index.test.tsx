// libraries
import { render, fireEvent } from '@testing-library/react'
import * as renderer from 'react-test-renderer'
import { ThemeProvider } from '@mui/material/styles'

// theme
import theme from '../../themes'

// components
import Checkbox from '../Checkbox'

describe('Check box component', () => {
  const onChange = jest.fn()

  const CheckboxComponent = ({ disable = false, isChecked = false }) => (
    <ThemeProvider theme={theme}>
      <Checkbox label='Default' onChange={onChange} disable={disable} isChecked={isChecked} />
    </ThemeProvider>
  )

  test('It should matches snapshot', () => {
    const checkbox = renderer.create(<CheckboxComponent />).toJSON()
    expect(checkbox).toMatchSnapshot()
  })

  test('It should render correctly', () => {
    const { getByText } = render(<CheckboxComponent />)
    expect(getByText('Default')).toBeInTheDocument()
  })

  test('On click function should be called once when check the checkbox', () => {
    const { getByRole } = render(<CheckboxComponent />)
    fireEvent.click(getByRole('checkbox'))
    expect(onChange).toHaveBeenCalledTimes(1)
  })

  test('Input should be disable', () => {
    const { getByRole } = render(<CheckboxComponent disable={true} />)
    expect(getByRole('checkbox')).toBeDisabled()
  })

  test('Input should not be checked by default', () => {
    const { getByRole } = render(<CheckboxComponent />)
    expect(getByRole('checkbox')).toHaveProperty('checked', false)
  })

  test('Input should be checked', () => {
    const { getByRole } = render(<CheckboxComponent isChecked={true} />)
    expect(getByRole('checkbox')).toHaveProperty('checked', true)
  })
})
