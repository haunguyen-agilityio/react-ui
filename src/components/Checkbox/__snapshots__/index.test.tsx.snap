// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check box component It should matches snapshot 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-8dopei-MuiFormControlLabel-root"
>
  <span
    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-mto4qr-MuiButtonBase-root-MuiCheckbox-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      data-indeterminate={false}
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SquareOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 3v18h18V3H3zm16 16H5V5h14v14z"
      />
    </svg>
  </span>
  <span
    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1e6sop9-MuiTypography-root"
  >
    Default
  </span>
</label>
`;
